samples/nrf_desktop/src/events/led_event.c

```diff
@@ -14,16 +14,29 @@ static const char * const mode_name[] = {
 #undef X
 };
 
-static void print_event(const struct event_header *eh)
+static size_t log_led_event(const struct event_header *eh, char *buf,
+			  size_t buf_len)
 {
 	struct led_event *event = cast_led_event(eh);
+	size_t temp;
+	size_t pos = 0;
+
+	temp = snprintf(buf, buf_len, "led_id:%u mode:%s period:%u color: <",
+			event->led_id, mode_name[event->mode], event->period);
+	pos += temp;
+	__ASSERT_NO_MSG((pos < buf_len) && (temp > 0));
```

Comment:
I am not fan of asserts in a code like this. Note that log messages may be used in release code where asserts are disabled.
Please handle the error case (e.g. stop adding strings where there is no place in the buffer; you can add `...` at the end if there was an overflow).